hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+  other_reset_list: [ "rst_core_ni" ],
+  bus_device: "tlul",
+  bus_host: "none",
+  regwidth: "32",
+  param_list: [
+    { name: "ByteOrder",
+      desc: '''Byte order to use when transmitting or receiving data.  If ByteOrder = 0,
+               the IP uses a Big-Endian ordering for the bytes in TXDATA and RXDATA.
+               The most significant byte (MSB) of TXDATA is transmitted first, and
+               received data is placed in the MSB location of RXDATA.  If ByteOrder = 1,
+               a Little-Endian ordering is used for these registers, and the LSB of each
+               gets priority for receiving and transmitting data.'''
+      type: "int",
+      default: "0"
+    },
+    { name: "MaxCS",
+      desc: "The number of active-low chip select (cs_n) lines to create. (Maximum 8)",
+      type: "int",
+      default: "8"
+    },
+  ],
+  available_output_list: [
+    { name:  "sck_o"
+      desc:  "SPI Clock"
+    },
+    { name:  "cs_no"
+      desc:  '''Chip Enable# (One hot, active low).  The size of this port should match MaxCS'''
+      width: "8"
+    },
+    { name:  "sd_o",
+      desc:  "SPI data IO outputs",
+      width: "4"
+    },
+    { name:  "sd_t",
+      desc:  "SPI data tri-state control",
+      width: "4"
+    }
+  ],
+  available_input_list: [
+    { name:  "sd_i",
+      desc:  "SPI data IO inputs",
+      width: "4"
+    },
+  ],
+  interrupt_list: [
+    { name: "error",
+      desc: '''Error-related interrupts, see ERROR_MASK register for more
+               information.'''
+    },
+    { name: "event",
+      desc: '''Event-related interrupts, see EVENT_MASK register for more
+               information.'''
+    }
+  ],
+  inter_signal_list: [
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_sck_i",
+      act:     "rcv",
+      width:   1,
+      package: ""
+    },
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_cs_ni",
+      act:     "rcv",
+      width:   8,
+      package: ""
+    },
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_sdo_i",
+      act:     "rcv",
+      width:   4,
+      package: ""
+    },
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_sdt_i",
+      act:     "rcv",
+      width:   4,
+      package: ""
+    },
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_sdi_o",
+      act:     "req",
+      width:   4,
+      package: ""
+    },
+  ]
+  registers: [
+    { name: "CONTROL",
+      desc: "Control register",
+      swaccess: "wo",
+      hwaccess: "hro",
+      fields: [
+        { bits: "31",
+          name: "SPIEN",
+          desc: '''Enables the SPI host.  On reset, this field is 0, meaning that no transactions can proceed.'''
+          resval: "0x0"
+        },
+        { bits: "30",
+          name: "RST_FSM",
+          desc: '''This bit holds the control FSM in reset when set to 1.'''
+          resval: "0x0",
+          hwqe: "true",
```

Comment:
The hwqe flag is applied per-register, not per-field (reggen spits out a warning about this, but it's all a bit chatty, so easy to miss)

Reply:
Flag Removed, as it's not actually needed in the latest vision of this IP.